packages/runtime/spec/reflection-contains-message-type.spec.ts

```diff
@@ -0,0 +1,42 @@
+import {MessageType, ScalarType, containsMessageType, getMessageType} from '../src';
+
+describe('containsMessageType', () => {
+    interface MyMessage {
+        stringField: string;
+    }
+
+    const MyMessage: MessageType<MyMessage> = new MessageType<MyMessage>('.test.MyMessage', [
+        {no: 1, name: 'string_field', kind: "scalar", T: ScalarType.STRING},
+    ]);
+
+    it('allows to extract type after .create', () => {
+        const msg = MyMessage.create({
+            stringField: "hello world",
+        });
+
+        if (containsMessageType(msg)) {
+            expect(getMessageType(msg)).toEqual(MyMessage);
+        } else {
+            fail("containsMessageType() must return true");
+        }
+    });
+
+    it('allows to extract type after .clone', () => {
+        const msg = MyMessage.create({
+            stringField: "hello world",
+        });
+
+        const cloned = MyMessage.clone(msg);
+        if (containsMessageType(cloned)) {
+            expect(getMessageType(cloned)).toEqual(MyMessage);
```

Comment:
With the typing change suggested below:

```suggestion
import {MessageType, ScalarType, containsMessageType, MESSAGE_TYPE} from '../src';

describe('containsMessageType', () => {
    interface MyMessage {
        stringField: string;
    }

    const MyMessage: MessageType<MyMessage> = new MessageType<MyMessage>('.test.MyMessage', [
        {no: 1, name: 'string_field', kind: "scalar", T: ScalarType.STRING},
    ]);

    it('allows to extract type after .create', () => {
        const msg = MyMessage.create({
            stringField: "hello world",
        });

        if (containsMessageType(msg)) {
            expect(msg[MESSAGE_TYPE]).toEqual(MyMessage);
        } else {
            fail("containsMessageType() must return true");
        }
    });

    it('allows to extract type after .clone', () => {
        const msg = MyMessage.create({
            stringField: "hello world",
        });

        const cloned = MyMessage.clone(msg);
        if (containsMessageType(cloned)) {
            expect(cloned[MESSAGE_TYPE]).toEqual(MyMessage);
```